service/api/prettyprint.go

```diff
@@ -403,14 +416,40 @@ func PrettyExamineMemory(address uintptr, memArea []byte, format byte) string {
 
 	var b strings.Builder
 	w := tabwriter.NewWriter(&b, 0, 0, 3, ' ', 0)
+
 	for i := 0; i < rows; i++ {
 		fmt.Fprintf(w, addrFmt, address)
-		for j := 0; j < cols && i*cols+j < l; j++ {
-			fmt.Fprintf(w, colFormat, memArea[i*cols+j])
+
+		for j := 0; j < cols; j++ {
+			offset := i*(cols*colBytes) + j*colBytes
+			if offset+colBytes <= len(memArea) {
+				data := memArea[offset : offset+colBytes]
+				fmt.Fprintf(w, colFormat, byteArrayToUInt(data))
+			}
 		}
 		fmt.Fprintln(w, "")
 		address += uintptr(cols)
 	}
 	w.Flush()
 	return b.String()
 }
+
+func byteArrayToUInt(arr []byte) interface{} {
+	reverse(arr)
```

Comment:
The right way of doing this is with shift and addition, or encoding/binary.ReadUintX

Reply:
aarzilli, I use shift and addition to reimplement this.

But I have another problem: how to determine the byte size? Here, I use little endian by default.
I just don't know how to determine the byte order in a core file.

Reply:
done